blaze/expr/reductions.py

```diff
@@ -14,6 +17,51 @@
 from .expressions import dshape_method_list, method_properties
 
 
+class Window(Expr):
+    __slots__ = (
+        '_hash',
+        '_child', '_group_by', '_sort_by', 'preceding', 'following',
+        '_asdshape'
+    )
+
+    __inputs__ = '_child', '_group_by', '_sort_by'
+
+    def __str__(self):
+        slots = '_group_by', '_sort_by', 'preceding', 'following'
+        return '%s(%s, %s)' % (
+            type(self).__name__,
+            self._child,
+            ', '.join('%s=%r' % (slot, getattr(self, slot)) for slot in slots)
+        )
+
+    def _dshape(self):
+        return self._asdshape
+
+    @property
+    def _name(self):
+        return self._child._name
+
+    def sort_by(self, key):
```

Comment:
I agree, mostly because of `list.sort`. I am just concerned with spelling this two different ways. Maybe we can deprecate `sort` on old expressions expressions and add `sort_by` for 0.9.0 and only implement `sort_by` for windows. This doesn't break any existing code and lets us move forward.